modules/network/src/main/java/org/apache/ignite/internal/network/direct/stream/DirectByteBufferStreamImplV1.java

```diff
@@ -1049,6 +1153,80 @@ public BitSet readBitSet() {
         return arr != null ? BitSet.valueOf(arr) : null;
     }
 
+    @Override
+    public ByteBuffer readByteBuffer() {
+        byte[] bytes;
+
+        switch (byteBufferState) {
+            case 0:
+                byteBufferFlag = readByte();
+
+                boolean isNull = (byteBufferFlag & BYTE_BYFFER_NOT_NULL_FLAG) == 0;
+
+                if (!lastFinished || isNull) {
+                    return null;
+                }
+
+                byteBufferState++;
+
+                //noinspection fallthrough
+            case 1:
+                if ((byteBufferFlag & BYTE_BYFFER_HAS_POSITION_FLAG) == 0) {
+                    byteBufferPosition = 0;
+                } else {
+                    byteBufferPosition = readInt();
+                }
+
+                if (!lastFinished) {
```

Comment:
maybe this `if` should be moved to closer to the `readInt` call

Reply:
Ok